core/services/vrf/integration_v2_test.go

```diff
@@ -917,6 +917,99 @@ func TestVRFV2Integration_SingleConsumer_NeedsTopUp(t *testing.T) {
 	assertNumRandomWords(t, consumerContract, numWords)
 }
 
+func TestVRFV2Integration_SingleConsumer_BigGasCallback_Sandwich(t *testing.T) {
+	config, db := heavyweight.FullTestDB(t, "vrfv2_singleconsumer_bigcallback_sandwich")
+	ownerKey := cltest.MustGenerateRandomKey(t)
+	uni := newVRFCoordinatorV2Universe(t, ownerKey, 1)
+	app := cltest.NewApplicationWithConfigAndKeyOnSimulatedBlockchain(t, config, uni.backend, ownerKey)
+	config.Overrides.GlobalEvmGasLimitDefault = null.NewInt(5e6, true)
+	config.Overrides.GlobalMinIncomingConfirmations = null.IntFrom(2)
+	consumer := uni.vrfConsumers[0]
+	consumerContract := uni.consumerContracts[0]
+	consumerContractAddress := uni.consumerContractAddresses[0]
+
+	// Create a subscription and fund with 5 LINK.
+	subID := subscribeAndAssertSubscriptionCreatedEvent(t, consumerContract, consumer, consumerContractAddress, big.NewInt(3e18), uni)
```

Comment:
Instead of `big.NewInt(3e18)` you can use `assets.Ether(3)`. Perhaps we should make a parallel set of methods for Juels, GJuels and LINK to make all of these tests more readable (in a separate PR).